src/player.cpp

```diff
@@ -2081,6 +2081,90 @@ void Player::death(Creature* lastHitCreature)
 		onIdleStatus();
 		sendStats();
 	}
+
+	despawn();
+}
+
+bool Player::spawn()
+{
+	setDead(false);
+
+	if (!g_game.map.placeCreature(getLoginPosition(), this, false, true)) {
+		return false;
+	}
+
+	const Position& pos = getPosition();
+
+	SpectatorVec spectators;
+	g_game.map.getSpectators(spectators, pos, true);
+	for (Creature* spectator : spectators) {
+		if (Player* tmpPlayer = spectator->getPlayer()) {
+			tmpPlayer->sendCreatureAppear(this, pos, false);
+		}
+	}
+
+	for (Creature* spectator : spectators) {
+		spectator->onCreatureAppear(this, false);
+	}
+
+	getParent()->postAddNotification(this, nullptr, 0);
+	g_game.addCreatureCheck(this);
+
+	addList();
+	return true;
+}
+
+void Player::despawn()
+{
+	if (isDead()) {
+		return;
+	}
+
+	sendCancelWalk();
```

Comment:
With `sendCancelWalk()` this no longer happens:
![GIF 09-01-2021 03-09-38 p  m](https://user-images.githubusercontent.com/28090948/104107876-53d56c80-5296-11eb-8159-8cd1b724be15.gif)
...
BUT

you need to add this:
```c++
onWalkComplete();
stopEventWalk();
```

otherwise the character will not be able to move when he reappears.

Reply:
This was tested and is working without any additional code. When you die while walking, your last step is automatically cancelled by server, sendCancelWalk is just there to sync client with current position.

Reply:
> This was tested and is working without any additional code. When you die while walking, your last step is automatically cancelled by server, sendCancelWalk is just there to sync client with current position.

Have you tried it on windows?
I got an error even with `sendCancelWalk` once the player appeared, it could not move, I did the tests by sending the player to a far away place with a click, and in the middle of the way it made him die, and then that happened

Reply:
You may be right, calling the method to clean auto walk directions shouldn't change much and the dir vector should be cleared, thanks.